src/onnx/quantize_dequantize_linear.cpp

```diff
@@ -0,0 +1,139 @@
+namespace onnx {
+
+void transform_quantize_dequantize_linear_inputs(const onnx_parser::node_info& info,
+                                                 const std::string& op_name,
+                                                 int block_size,
+                                                 int axis,
+                                                 std::vector<instruction_ref>& args)
+{
+    const auto x      = args.at(0);
+    const auto x_lens = x->get_shape().lens();
+    const auto x_rank = x_lens.size();
+
+    instruction_ref y_scale = args.at(1);
+    const auto y_scale_lens = y_scale->get_shape().lens();
+    const auto y_scale_rank = y_scale_lens.size();
+
+    // Per-tensor (per-layer) granularity
+    if(y_scale->get_shape().elements() == 1)
+    {
+        std::transform(args.begin() + 1, args.end(), args.begin() + 1, [&](auto ins) {
+            return info.add_instruction(make_op("multibroadcast", {{"out_lens", x_lens}}), ins);
+        });
+    }
+    // Per-axis granularity
+    else if(y_scale_rank == 1)
+    {
+        axis = tune_axis(x_rank, axis, op_name);
+        if(x_lens[axis] != y_scale_lens[0])
+        {
+            MIGRAPHX_THROW(op_name + ": For per axis granularity the length of y_scale (actual: " +
+                           to_string(y_scale_lens[0]) + ") must be equal to size of x on axis " +
+                           to_string(axis) + "(actual :" + to_string(x_lens[axis]) + ")");
+        }
+
+        std::transform(args.begin() + 1, args.end(), args.begin() + 1, [&](auto ins) {
+            return info.add_instruction(
+                make_op("broadcast", {{"axis", axis}, {"out_lens", x_lens}}), ins);
+        });
+    }
+    // Blocked granularity
+    else
+    {
+        axis = tune_axis(x_rank, axis, op_name);
+        if(block_size == 0)
+        {
+            MIGRAPHX_THROW(op_name + ": Invalid blocksize(0)");
+        }
+
+        if(x_rank != y_scale_rank)
+        {
+            MIGRAPHX_THROW(op_name + ": x(rank: " + to_string(x_rank) +
+                           ") and y_scale(rank: " + to_string(y_scale_rank) +
+                           ") must be of same rank for block granularity");
+        }
+
+        for(auto i = 0u; i < x_lens.size(); ++i)
+        {
+            if(x_lens[i] != y_scale_lens[i] and i != axis)
+            {
+                MIGRAPHX_THROW(op_name + ": x(shape: " + to_string_range(x_lens) +
+                               ") and y_scale(shape: " + to_string_range(y_scale_lens) +
+                               ") shapes may only differ along provided axis(" + to_string(axis) +
+                               ")");
+            }
+        }
+
+        // Given x shape (D0, ..., Di, ..., Dn), y_scale shape (S0, ... Si, ...Sn) and
+        // axis=i, the accepted range is [ceil(Di/Si), ceil(Di/(Si-1))-1]
+        float di           = x_lens[axis];
+        float si           = y_scale_lens[axis];
+        int block_size_min = std::ceil(di / si);
```

Comment:
Sample code that can be added below if exception above is removed -- for `block_size == 0`.

`if(block_size ==0) block_size = block_size_min;`